darts/preprocessing/base_transformer.py

```diff
@@ -0,0 +1,238 @@
+
+        self._reversible = reversible
+        self._can_predict = can_predict
+        self._fittable = fittable
+
+    @property
+    def reversible(self) -> bool:
+        """
+        Returns
+        -------
+        bool
+            Whether transformer has inverse_transform.
+        """
+        return self._reversible
+
+    @property
+    def can_predict(self) -> bool:
+        """
+        Returns
+        -------
+        bool
+            Whether transformer can be used to predict.
+        """
+        return self._can_predict
+
+    @property
+    def fittable(self) -> bool:
+        """
+        Returns
+        -------
+        bool
+            Whether transformer can be fitted.
+        """
+        return self._fittable
+
+    @property
+    def name(self):
+        """
+        Returns
+        -------
+        str
+            Name of transformer.
+        """
+        return "BaseTransformer"
+
+    def validate(self, data: T) -> bool:
+        """
+        Validate data using validators set at init. If validation will be unsuccessful it will raise exception
+        potentially with reason/explanation why validation failed.
+
+        Parameters
+        ----------
+        data
+            Object on which validation functions will be run.
+
+        Returns
+        -------
+        True
+            If successful.
+        """
+        for validator in self._validators:
+            reason = validator.reason if validator.reason else ""
+            raise_if_not(validator(data), f"Validation failed for {self.name}\n{reason}", logger)
+        return True
+
+    @abstractmethod
+    def transform(self, data: T, *args, **kwargs) -> T:
+        """
+        Transform data. Not implemented in base class and has to be implemented by deriving classes.
+
+        Parameters
+        ----------
+        data
+            Object which will be transformed.
+        args
+            Positional arguments for the `transform` method
+        kwargs
+            Keyword arguments for the `transform` method
+
+        Returns
+        -------
+        T
+            Transformed data.
+        """
+        raise NotImplementedError(f"transform not implemented for transformer {self.name}")
+
+    def inverse_transform(self, data: T, *args, **kwargs) -> T:
+        """
+        Inverse transformation of data. Not implemented in base class.
+
+        Parameters
+        ----------
+        data
+            Object which will be inverse transformed.
+        args
+            Positional arguments for the `inverse_transform` method
+        kwargs
+            Keyword arguments for the `inverse_transform` method
+
+        Returns
+        -------
+        T
+            Inverse transformed data.
+        """
+        raise NotImplementedError(f"inverse_transform not implemented for transformer {self.name}")
+
+    def predict(self, data: T, *args, **kwargs) -> T:
+        """
+        Predict function. Not implemented in base class.
```

Comment:
Can you give an example of a transformer that has a `predict` method? Maybe an additional sentence in the docstring that explains the role of the `predict` functions could be useful, since it's not immediately clear to me

Reply:
Something to put on radar, but here should be some discusion how do we want to treat pipeline with predict at last "transformer". Do we run transform and all that stuff on it or just prediction? More detailed documentation is also valid point. Will add it

Reply:
I am not a big fan of having a `predict` method. I believe it would make the API cleaner to restrict transformers and pipelines to preprocessing only, and then call `model.predict()` on the transformed data